apps/whale-api/src/module.api/cache/defid.cache.ts

```diff
@@ -104,13 +109,19 @@ export class DeFiDCache extends GlobalCache {
     )
   }
 
-  private async fetchPoolPairInfo (id: string): Promise<PoolPairInfo | undefined> {
+  /**
+   * Retrieve poolPair info via rpc client
+   * @param {string} idOrSymbol - id or symbol
+   * @return {PoolPairInfoWithId | undefined}
+   */
+  private async fetchPoolPairInfo (idOrSymbol: string): Promise<PoolPairInfoWithId | undefined> {
     try {
-      const result = await this.rpcClient.poolpair.getPoolPair(id)
-      if (result[id] === undefined) {
-        return undefined
+      const result = await this.rpcClient.poolpair.getPoolPair(idOrSymbol)
+      const [id, poolPairInfo] = Object.entries(result)[0]
+      return {
```

Comment:
This no longer seem to be checking for potential `undefined` results. Considering this seems deliberate this is most likely as per design, but wanted to point this out in case this may need to be addressed.

Reply:
hmm.. i think its okay as its wrapped inside a `try..catch` for rpc call, if getting `not found` from rpc will return undefined, else throw err